### ISSUE TITLE
Can't load a model from private repo with binaries stored with DVC

### ISSUE THREAD

**Issue by aguschin** [iterative/mlem#528]
I've set `GITHUB_TOKEN` and `GITHUB_USERNAME` #91 

Then:
```shell
$ mlem --tb clone models/rf --project https://github.com/aguschin/example-mlem-private cloned
⏳️ Loading meta from https://github.com/aguschin/example-mlem-private/tree/dvc/models/rf.mlem
🐏 Cloning https://github.com/aguschin/example-mlem-private/tree/dvc/models/rf.mlem
Traceback (most recent call last):
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/scmrepo/git/backend/dulwich/__init__.py", line 200, in clone
    repo = clone_from()
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/dulwich/porcelain.py", line 551, in clone
    return client.clone(
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/dulwich/client.py", line 760, in clone
    result = self.fetch(path, target, progress=progress, depth=depth)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/dulwich/client.py", line 837, in fetch
    result = self.fetch_pack(
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/dulwich/client.py", line 2076, in fetch_pack
    refs, server_capabilities, url = self._discover_references(
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/dulwich/client.py", line 1934, in _discover_references
    resp, read = self._http_request(url, headers)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/dulwich/client.py", line 2216, in _http_request
    raise HTTPUnauthorized(resp.getheader("WWW-Authenticate"), url)
dulwich.client.HTTPUnauthorized: No valid credentials provided

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/dvc/scm.py", line 145, in clone
    git = Git.clone(url, to_path, progress=pbar.update_git, **kwargs)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/scmrepo/git/__init__.py", line 143, in clone
    backend.clone(url, to_path, **kwargs)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/scmrepo/git/backend/dulwich/__init__.py", line 203, in clone
    raise CloneError(url, to_path) from exc
scmrepo.exceptions.CloneError: Failed to clone repo 'https://github.com/aguschin/example-mlem-private' to '/var/folders/tv/l60j0x050p536g3bh8g2w1n80000gn/T/tmpl32vv8x3dvc-clone'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/bin/mlem", line 8, in <module>
    sys.exit(app())
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/typer/main.py", line 328, in __call__
    raise e
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/typer/main.py", line 311, in __call__
    return get_command(self)(*args, **kwargs)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/typer/core.py", line 778, in main
    return _main(
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/typer/core.py", line 216, in _main
    rv = self.invoke(ctx)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/aguschin/Git/iterative/mlem/mlem/cli/main.py", line 174, in invoke
    return super().invoke(ctx)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/typer/main.py", line 683, in wrapper
    return callback(**use_params)  # type: ignore
  File "/Users/aguschin/Git/iterative/mlem/mlem/cli/main.py", line 475, in inner
    f(*iargs, **ikwargs)
  File "/Users/aguschin/Git/iterative/mlem/mlem/cli/clone.py", line 28, in clone
    clone(
  File "/Users/aguschin/Git/iterative/mlem/mlem/telemetry.py", line 50, in inner
    return f(*args, **kwargs)
  File "/Users/aguschin/Git/iterative/mlem/mlem/api/commands.py", line 197, in clone
    return meta.clone(
  File "/Users/aguschin/Git/iterative/mlem/mlem/core/objects.py", line 547, in clone
    download = art.relative(
  File "/Users/aguschin/Git/iterative/mlem/mlem/core/artifacts.py", line 78, in materialize
    return self._download(target_path)
  File "/Users/aguschin/Git/iterative/mlem/mlem/contrib/dvc.py", line 85, in _download
    with self.open() as fin, open(target_path, "wb") as fout:
  File "/Users/aguschin/.pyenv/versions/3.9.5/lib/python3.9/contextlib.py", line 117, in __enter__
    return next(self.gen)
  File "/Users/aguschin/Git/iterative/mlem/mlem/contrib/dvc.py", line 97, in open
    with open(
  File "/Users/aguschin/.pyenv/versions/3.9.5/lib/python3.9/contextlib.py", line 117, in __enter__
    return next(self.gen)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/dvc/api/data.py", line 197, in _open
    with Repo.open(repo, rev=rev, subrepos=True, uninitialized=True) as _repo:
  File "/Users/aguschin/.pyenv/versions/3.9.5/lib/python3.9/contextlib.py", line 117, in __enter__
    return next(self.gen)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/dvc/external_repo.py", line 39, in external_repo
    path = _cached_clone(url, rev, for_write=for_write)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/dvc/external_repo.py", line 169, in _cached_clone
    clone_path, shallow = _clone_default_branch(url, rev, for_write=for_write)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/funcy/decorators.py", line 45, in wrapper
    return deco(call, *dargs, **dkwargs)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/funcy/flow.py", line 274, in wrap_with
    return call()
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/funcy/decorators.py", line 66, in __call__
    return self._func(*self._args, **self._kwargs)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/dvc/external_repo.py", line 239, in _clone_default_branch
    git = clone(url, clone_path)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.9/site-packages/dvc/scm.py", line 150, in clone
    raise CloneError(str(exc))
dvc.scm.CloneError: Failed to clone repo 'https://github.com/aguschin/example-mlem-private' to '/var/folders/tv/l60j0x050p536g3bh8g2w1n80000gn/T/tmpl32vv8x3dvc-clone'
```

I can share access to https://github.com/aguschin/example-mlem-private to ease the fixing process.

**Comment by mike0sv** [comment]
I think the error is raised because when you use http url, mlem uses github fs with env for auth. but dvc still uses ssh credentials and fails. so if you use dvc with mlem, you need both. or use ssh auth and url with mlem

**Comment by aguschin** [comment]
Yes, the problem was with that. We need to write to docs about this. That you need both TOKEN auth (for MLEM) and SSH AUTH (`gh auth login` for example) to work with private DVC repos.

**Comment by aguschin** [comment]
so, I tried to make a fix in #596. Need to see if that's working, otherwise we need to update docs with mlem.ai#312

**Comment by aguschin** [comment]
Ok, looks like it didn't work out. One still need to do both authorizations.
